python/TestHarness/schedulers/QueueManager.py

```diff
@@ -54,30 +55,36 @@ def handleQueueLaunch(self, tester, output):
     def handleQueueStatus(self, tester, output):
         return
 
+    # convert a dictionary_bucket status to a tester_bucket status
+    def createStatusBucket(self, json_status):
+        # Create a status_bucket according to what is stored in the json file
+        test_status = namedtuple('status_bucket', json_status)
+        return test_status(status=json_status['status'], color=json_status['color'])
+
     # Update tester status in queue_data
     def updateTesterStatus(self, tester):
-        # Update the queue_data with new relevant information
         self.putData(self.getJobName(tester),
-                     job_name=self.getJobName(tester),
-                     status=tester.getStatusMessage(),
-                     skipped=tester.isSkipped(),
-                     is_pending=tester.isPending(),
-                     is_silent=tester.isSilent(),
-                     is_deleted=tester.isDeleted(),
-                     did_pass=tester.didPass(),
-                     did_fail=tester.didFail(),
-                     did_diff=tester.didDiff(),
-                     test_dir=self.getWorkingDir(tester),
-                     test_name=tester.getTestName(),
-                     input_name=self.options.input_file_name)
+                     job_name       = self.getJobName(tester),
+                     skipped        = tester.isSkipped(),
+                     is_pending     = tester.isPending(),
+                     is_silent      = tester.isSilent(),
+                     is_deleted     = tester.isDeleted(),
+                     did_pass       = tester.didPass(),
+                     did_fail       = tester.didFail(),
+                     did_diff       = tester.didDiff(),
+                     test_dir       = self.getWorkingDir(tester),
+                     test_name      = tester.getTestName(),
+                     input_name     = self.options.input_file_name,
+                     status_message = tester.getStatusMessage(),
+                     status_bucket  = dict(tester.getStatus()._asdict()))
```

Comment:
because I am now storing the actual status_bucket in the dictionary, we can consider removing all those individual test statuses from the dictionary (is_pending, did_pass, skipped, etc). However, because its a bit more expensive and code intensive to; create a bucket, and then compare that bucket to a tester bucket to figure out the same results as doing a:  `if dict['is_pending']:` I think we should leave this alone.